include/envoy/filesystem/filesystem.h

```diff
@@ -102,6 +102,13 @@ class Instance {
    */
   virtual std::string fileReadToEnd(const std::string& path) PURE;
 
+  /**
+   * @path full file path on input which is truncated to the path
+   * @name the resulting file name component from the input path
+   */
+  virtual std::pair<absl::string_view, absl::string_view>
```

Comment:
nit: prefer returning a struct here with named members. It's not clear what is being returned.

Reply:
Would it be acceptable to keep the `std::pair` if we also improve the documentation of the method and explicitly describe the return value? Since the method name is relatively descriptive itself?

Reply:
Please just swap it out. It's a trivial change that greatly increases readability, especially for an abstract API.